### DISCUSSION TITLE
Messaging Policy

### DISCUSSION THREAD

**Discussion by rnugent3** [HydrologicEngineeringCenter/HEC-FDA#355] [Ideas]
IErrorMessage enumeration is defined like this. Let's display to the user all major, fatal, and severe errors, and write all errors to file. 

Question for @HenryGeorgist: is severe meant to be worse than fatal? If I have the choice between a severe illness and a fatal illness, I'm picking severe illness. 

```
    public enum ErrorLevel : byte
    {
        Unassigned = 0X00,
        Info = 0x01,
        Minor = 0x02,
        Major = 0x04,
        Fatal = 0x20,
        Severe = 0x40,

    }

**Comment by HenryGeorgist** [comment]
it should not be significant effort to make fatal worse than severe, just switch their order and values. The only issue may be in some of the consuming logic where severe was being filtered earlier, now fatal errors will also get included in the filter.

**Reply by rnugent3** [reply]
I thought maybe this was a weird software thing, like setting the index of the first position of an array with 0. 

I am happy to work with it as is.

**Comment by HenryGeorgist** [comment]
it is written as a bit flag so that we can use filters like  if error>=filterlevel and we will get the result we want... so if you want fatal to be worse than severe you should correct it.
